File: WellnessWingman/Models/EntryAnalysis.cs

namespace HealthHelper.Models;

public class EntryAnalysis
{
    public int AnalysisId { get; set; }
    public int EntryId { get; set; }
    public Guid? ExternalId { get; set; }
    public string ProviderId { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public DateTime CapturedAt { get; set; }
    public string InsightsJson { get; set; } = string.Empty;

    /// <summary>
    /// Version of the JSON schema used in InsightsJson.
    /// Enables schema evolution and backward compatibility.
    /// </summary>
    public string SchemaVersion { get; set; } = "1.0";
}

File: kotlin/shared/src/commonMain/kotlin/com/wellnesswingman/data/model/EntryAnalysis.kt
package com.wellnesswingman.data.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

/**
 * Represents an LLM analysis of a tracked entry.
 */
@Serializable
data class EntryAnalysis(
    val analysisId: Long = 0,
    val entryId: Long,
    val externalId: String? = null,
    val providerId: String = "",
    val model: String = "",
    val capturedAt: Instant,
    val insightsJson: String = "",

    /**
     * Version of the JSON schema used in insightsJson.
     * Enables schema evolution and backward compatibility.
     */
    val schemaVersion: String = "1.0"
)
